posthog/api/early_access_feature.py

```diff
@@ -61,6 +61,50 @@ class Meta:
         ]
         read_only_fields = ["id", "feature_flag", "created_at"]
 
+    def update(self, instance: EarlyAccessFeature, validated_data: Any) -> EarlyAccessFeature:
+        stage = validated_data.get("stage", None)
+
+        if stage == EarlyAccessFeature.Stage.BETA:
+
+            super_conditions = lambda feature_flag_key: [
+                {
+                    "properties": [
+                        {
+                            "key": f"$feature_enrollment/{feature_flag_key}",
+                            "type": "person",
+                            "operator": "exact",
+                            "value": ["true"],
+                        },
+                    ],
+                    "rollout_percentage": 100,
+                },
+            ]
+
+            related_feature_flag = instance.feature_flag
+            if related_feature_flag:
+
+                related_feature_flag_key = related_feature_flag.key
+                serialized_data_filters = {
+                    **related_feature_flag.filters,
+                    "super_groups": super_conditions(related_feature_flag_key),
+                }
+
+                serializer = FeatureFlagSerializer(
+                    related_feature_flag, data={"filters": serialized_data_filters}, context=self.context, partial=True
+                )
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+        elif instance.stage == EarlyAccessFeature.Stage.BETA:
```

Comment:
this check applies on every update, so if I updated something that wasn't stage, it will still remove the super group?

Reply:
good catch, was trying to account for updates that have nothing to do with the stage but ended up with bad logic